### PR TITLE
Change 'registration' to 'sign-up' in the context of registration in SIO

### PR DESCRIPTION
**Pull request by Igifigi**
SIO users often did not distinguish between contest registration and system registration. For this reason, registration for the SIO was changed to sign-up.

### CODE REVIEW

File: oioioi/base/registration_backend.py [hunk 1]
```diff
@@ -63,18 +63,18 @@ def register(self, form):
...
 urlpatterns = [
-    re_path(r'^register/$', RegistrationView.as_view(), name='registration_register'),
+    re_path(r'^sign-up/$', RegistrationView.as_view(), name='sign-up'),
 ]
 
 if not settings.SEND_USER_ACTIVATION_EMAIL:
     urlpatterns += [
         re_path(
-            r'^register/complete/$',
+            r'^sign-up/complete/$',
```
Comment (DietPawel) [inline oioioi/base/registration_backend.py:72]: I would like to see some redirection from previous register to sign-up just not to broke already existing links in some content

File: oioioi/base/registration_backend.py [hunk 2]
```diff
@@ -63,18 +63,18 @@ def register(self, form):
 
 
 urlpatterns = [
-    re_path(r'^register/$', RegistrationView.as_view(), name='registration_register'),
+    re_path(r'^sign-up/$', RegistrationView.as_view(), name='sign-up'),
```
Comment (DietPawel) [inline oioioi/base/registration_backend.py:66]: I am not sure it is bad, just thinking if `sign-up` is unique enough

**Review by DietPawel — CHANGES_REQUESTED** [review]

File: oioioi/base/registration_backend.py (hunk 2 above)
Comment (A-dead-pixel) [reply oioioi/base/registration_backend.py:66]: I think that `create-account` will be better and sufficient. Also, then the Polish translation would be different from "rejestracja".

Comment (DietPawel) [reply oioioi/base/registration_backend.py:66]: I would like to say I only disliked url's name, not the url itself. As for the name I like *create account*, as it contains a noun making it more specific. In terms of translations which is the main point of this PR I consider "załóż konto" as good translation of "sign-up". Anything that won't be the same two words for making an account and enrolling to the contest will fix it for me.

File: oioioi/base/registration_backend.py [hunk 3]
```diff
@@ -63,17 +63,23 @@ def register(self, form):
...
+    re_path(r'^register/$', RedirectView.as_view(url='/sign-up/', permanent=True), name='registration_register'),
```
Comment (DietPawel) [inline oioioi/base/registration_backend.py:67]: Again, I am picky, but `registration_redirect` is so much better than `registration_register` in my opinion

File: oioioi/base/registration_backend.py [hunk 4]
```diff
@@ -63,17 +63,23 @@ def register(self, form):
...
         re_path(
-            r'^register/complete/$',
+            r'^sign-up/complete/$',
             TemplateView.as_view(
                 template_name='registration/'
                 'registration_and_activation_complete.html'
             ),
+            name='sign-up_complete',
+        ),
+        re_path(
```
Comment (DietPawel) [inline oioioi/base/registration_backend.py:80]: I asked for it, but I must have marked the wrong line, sorry. I like the redirection from `/register/` to `/sign-up/`, but this complete link is a not necessary.

**Review by DietPawel — CHANGES_REQUESTED** [review]

**Review by DietPawel — APPROVED** [review]

**Pull request merged**